src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -103,7 +108,19 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Birthday updatedBirthday = editPersonDescriptor.getBirthday().orElse(personToEdit.getBirthday());
-        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
+        Set<Tag> updatedTags = new HashSet<>(editPersonDescriptor.getTags().orElse(personToEdit.getTags()));
+
+        Set<String> netWorthTags = Set.of("highnetworth", "midnetworth", "lownetworth");
+
+        Optional<Tag> newNetWorthTag = updatedTags.stream()
+                .filter(tag -> netWorthTags.contains(tag.tagName.toLowerCase()))
+                .findFirst();
+
+        if (newNetWorthTag.isPresent()) {
+            updatedTags.removeIf(tag -> netWorthTags.contains(tag.tagName.toLowerCase()));
+            updatedTags.add(newNetWorthTag.get());
+        }
```

Comment:
Good defensive coding to ensure that only one netWorth tag is present at any time